src/compas_fab/robots/robot.py

```diff
@@ -302,506 +464,696 @@ def get_RCF(self, group=None):
         """
         return self.get_base_frame(group)
 
-    def represent_frame_in_RCF(self, frame_WCF, group=None):
-        """Returns the representation of a frame in the world coordinate frame
-        (WCF) in the robot's coordinate frame (RCF).
-        """
-        frame_RCF = frame_WCF.transformed(self.transformation_WCF_RCF(group))
-        return frame_RCF
-
-    def represent_frame_in_transformed_RCF(self, frame_WCF, configuration, group=None):
+    def represent_frame_in_transformed_RCF(self, frame_WCF, configuration=None, group=None):
```

Comment:
I divided the method into 2 private methods `_get_current_base_frame` and `_get_current_transformation_WCF_RCF`, because they are actually only needed in `plan_motion`. `plan_motion` has a different behavior than `plan_cartesian_motion` and `inverse_kinematics` , which I only realized with the abb on a linear track:
Calculating a motion plan for the planning group 'abb' expects the goal_constraints to be in the current (possibly transformed) robot coordinate system. (base frame of abb moved along the linear axis). That is why the base frame has to be calculated with the start_configuration.